sys/include/net/gnrc/netapi.h

```diff
@@ -103,6 +103,44 @@ typedef struct {
     uint16_t data_len;          /**< size of the data / the buffer */
 } gnrc_netapi_opt_t;
 
+/**
+ * @brief   Shortcut function for sending @ref GNRC_NETAPI_MSG_TYPE_SND or
+ *          @ref GNRC_NETAPI_MSG_TYPE_RCV messages
+ *
+ * @param[in] pid       PID of the targeted network module
+ * @param[in] pkt       pointer into the packet buffer holding the data to send
+ * @param[in] type      type of the message to send. Must be either
+ *                      @ref GNRC_NETAPI_MSG_TYPE_SND or
+ *                      @ref GNRC_NETAPI_MSG_TYPE_RCV
+ *
+ * @return              1 if packet was successfully delivered
+ * @return              -1 on error (invalid PID or no space in queue)
+ */
+int _gnrc_netapi_send_recv(kernel_pid_t pid, gnrc_pktsnip_t *pkt, uint16_t type);
```

Comment:
I find it suboptimal that these two internal (private) functions are now exposed, as if they are part of the API, when this header is included. I would leave them hidden in the implementation (i.e. `gnrc_netapi.c`)

Reply:
This is purely semantics, but the actual API are the message types. Everything else below the `gnrc_netapi_opt_t` type definition is actually just helper functions (as are these).

Since the actual optimization lies - as pointed out by @gebart - in the exposure of this more complicated functions and wrapping the simpler functions around them, they need to be defined where the definitions of the wrapper functions can "see" them (i.e. in the same header or a header that is included into this header). If it really bugs you that yet another (more "raw") helper function is exposed, I can put it in something like `netapi/internal.h`, but then we also should revisit `core` for such things, because the same is done there as it is in `xtimer`.